R/utilities-mpi-error-handling.R

```diff
@@ -0,0 +1,153 @@
+#' @title checkSimulationLoaded
+#' @description Check that the simulation has been loaded on the core.
+#' @param simulation the simulation object to be checked if loaded successfully
+#' @param simulationFilePath the path to the simulation object to be checked if loaded successfully
+#' @export
+checkSimulationLoaded <- function(simulation, simulationFilePath) {
+  success <- FALSE
+  if (identical(simulation$sourceFile, simulationFilePath)) {
+    success <- TRUE
+  }
+  return(success)
+}
+
+#' @title checkLibraryLoaded
+#' @description Check that a library has been successfully loaded on the core.
+#' @param libraryName the library to be checked if loaded successfully
+#' @export
+checkLibraryLoaded <- function(libraryName) {
+  success <- FALSE
+  if (libraryName %in% .packages()) {
+    success <- TRUE
+  }
+  return(success)
+}
+
+
+#' @title checkNotNull
+#' @description Check that an object is not null.
+#' @param object to be checked if not null
+#' @export
+checkNotNull <- function(object) {
+  success <- FALSE
+  if (!is.null(object)) {
+    success <- TRUE
+  }
+  return(success)
+}
+
+#' @title checkAllCoresSuccessful
+#' @description Check that all cores executed an mpi.remote.exec command successfully.
+#' @param coreResults list of results returned by each core after an mpi.remote.exec command is complete
+checkAllCoresSuccessful <- function(coreResults) {
+  allCoreResults <- sapply(1:length(coreResults), function(n, obj) obj[[n]], coreResults)
```

Comment:
Can you give an example of how this will be called? I don't understand the syntax

Reply:
mpi.remote.exec sends a command to each core and returns the combined results of the command from each core in a list.  I modified the methods that are called by mpi.remote.exec on each core to return true if successfully completed.   This function just checks that all returned values in the list are TRUE.

Reply:
I get what it does. I don't understand the inplemenation. Sapply works with an iterrable directly. Why do you need to do 1: lengths to retrieve each single item?

Reply:
Simplified this